boot/zephyr/main.c

```diff
@@ -128,6 +128,14 @@ static void do_boot(struct boot_rsp *rsp)
     vt = (struct arm_vector_table *)(flash_base +
                                      rsp->br_image_off +
                                      rsp->br_hdr->ih_hdr_size);
+
+#ifdef CONFIG_CPU_CORTEX_M_SOMETHING
```

Comment:
There's probably a way of testing if the features, instruct/data cache, MPU, etc exist in a target MCU.

Reply:
Changed now!

Reply:
@utzig I guess the CMSIS needs to provide the information by enabling the __ICACHE_PRESENT and __DCACHE_PRESENT macros. In the absence of definitions, the calls to these CMSIS functions are suppressed.

Reply:
Do you think I should add something like:
```#if defined(__ICACHE_PRESENT) && __ICACHE_PRESENT``` and ```#if defined(__DCACHE_PRESENT) && __DCACHE_PRESENT```?